openghg/transform/flux/_edgar.py

```diff
@@ -63,210 +204,96 @@ def parse_edgar(
-    else:
-        folder_filelist = list(datapath.glob("*"))
+    folder_filelist = [x for x in datapath.iterdir() if x.is_file()]
 
     # Extract netcdf files (only, for now) - ".txt" is also an option (not implemented)
-    suffix = ".nc"
-    data_files = [file for file in folder_filelist if file.suffix == suffix]
+    # Path(file.name).suffix workaround because zipfile.Path.suffix is Python 3.11+
+    data_files = [file for file in folder_filelist if pathlib.Path(file.name).suffix == ".nc"]
 
     if not data_files:
-        raise ValueError("Expect EDGAR '.nc' files." f"No suitable files found within datapath: {datapath}")
+        raise ValueError(f"No '.nc' files found in datapath: {datapath}")
 
-    for file in data_files:
-        try:
-            db_info = _extract_file_info(file)
-        except ValueError:
-            db_info = {}
-            continue
-
-    # Extract species from filename if not specified
-    try:
-        species_from_file: Optional[str] = db_info["species"]
-    except KeyError:
-        species_from_file = None
-
-    if species is None:
-        species = species_from_file
-
-    # Check synonyms and compare against filename value
-    if species is not None:
-        species_label = define_species_label(species)[0]
-        # species_label = synonyms(species).lower()
-    else:
-        raise ValueError("Unable to retrieve species from database filenames." " Please specify")
-
-    if species_from_file is not None and species_label != synonyms(species_from_file):
-        logger.warning(
-            "Input species does not match species extracted from",
-            " database filenames. Please check.",
-        )
-
-    # If version not yet found, extract version from file naming scheme
     if edgar_version is None:
-        possible_version = db_info["version"]
-        possible_version_clean = clean_string(possible_version)
-        if possible_version_clean in known_versions:
-            edgar_version = possible_version_clean
-
-    edgar_version = clean_string(edgar_version)
-
-    if edgar_version not in known_versions:
-        if edgar_version is None:
-            raise ValueError(
-                f"Unable to infer EDGAR version ({edgar_version})."
-                f" Please pass as an argument (one of {known_versions})"
-            )
-        else:
-            raise ValueError(
-                f"Unable to infer EDGAR version." f" Please pass as an argument (one of {known_versions})"
-            )
-
-    # TODO: May want to split out into a separate function, so we can use this for
-    # - yearly - "v6.0_CH4_2015_TOTALS.0.1x0.1.nc"
-    # - sectoral - "v6.0_CH4_2015_ENE.0.1x0.1.nc"
-    # - monthly sectoral - "v6.0_CH4_2015_1_ENE.0.1x0.1.nc", "v6.0_CH4_2015_2_ENE.0.1x0.1.nc", ...
-
-    if isinstance(date, int):
-        date = str(date)
+        # check if _readme.html is in folder_filelist, and if so, try to extract version
+        if (readme := datapath / "_readme.html").exists():
```

Comment:
If I was being a bit nitpicky - I think we should try and avoid more complex syntax like the walrus operator, `:=`, in favour of being a bit more verbose for clarity. I understand it can make syntax a bit neater but I just think it can make code a bit more confusing sometimes (and so can make it a bit harder to contribute).

Also Line 280 and 596.

Reply:
done